flashinfer/moe_ep/kernel_src/sm100/cutedsl_megamoe/src/moe_nvfp4_swapab/contract.py

```
@@ -309,10 +310,12 @@ class Contract:
     mapping: MappingSpec
 
     def __post_init__(self) -> None:
-        # Validate eagerly so malformed contracts fail at construction time.
-        self.mapping.normalize(domain=self.domain, codomain=self.codomain)
+        # Touching ``table`` validates eagerly (malformed contracts fail at
+        # construction time) and caches the normalized table so repeated
+        # property accesses don't re-run ``normalize``.
+        self.table
```

Ruff (0.16.0)

[warning] Found useless expression. Either assign it to a variable or remove it.

(B018)